### PR TITLE
Fix BigQuery: logging errors if exception is thrown during load_table_from_uri

### PR DESCRIPTION
**Pull request by joelluijmes**
…_from_uri

## Summary

Fixes logging of error when exception is thrown during BigQuery load_table_from_uri. Currently another exception is thrown due invalid handling.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/tasks/gcp/bigquery.py [hunk 1]
```diff
@@ -387,8 +387,9 @@ def run(
                 job_config=job_config,
             ).result()  # block until job is finished
```
Comment (zangell44) [inline src/prefect/tasks/gcp/bigquery.py:388]: I think it makes sense to split up the initialization and `result` calls here

```python
load_job = client.load_table_from_uri(
                uri,
                table_ref,
                location=location,
                job_config=job_config,
            )
load_job.result()

File: src/prefect/tasks/gcp/bigquery.py [hunk 2]
```diff
@@ -387,8 +387,9 @@ def run(
...
         except Exception as exception:
-            for error in load_job.errors:
-                self.logger(error)
+            for error in exception.errors:
+                self.logger.error(error)
```
Comment (zangell44) [inline src/prefect/tasks/gcp/bigquery.py:391]: After initializing `load_job` correctly (see above), `load_job.errors` might actually contain some useful information. See thread here https://github.com/googleapis/google-cloud-python/issues/6573.

Something like the following might work
```python
except Exception as exception:
            self.logger.error(exception)
            try:
                for error in load_job.errors:
                    self.logger.exception(error)
            except NameError:
                pass
            raise FAIL(exception) from exception
```

**Review by zangell44 — CHANGES_REQUESTED** [review]
Thanks for taking time to update the error handling @joelluijmes!

A couple suggestions to make sure we also capture additional error info if present

File: src/prefect/tasks/gcp/bigquery.py (hunk 2 above)
Comment (joelluijmes) [reply src/prefect/tasks/gcp/bigquery.py]: Ah good suggestion! I didn't actually dive in to which errors[] the message was referring. But I opted to log everything :)

File: src/prefect/tasks/gcp/bigquery.py [hunk 3]
```diff
@@ -385,10 +385,21 @@ def run(
...
+            )
+
+            # Start the job and wait for it to complete and get the result
+            load_job.result()
         except Exception as exception:
-            for error in load_job.errors:
-                self.logger(error)
+            self.logger.error(exception)
+
+            if exception.errors is not None:
```
Comment (zangell44) [inline src/prefect/tasks/gcp/bigquery.py:395]: I think we can remove this. I'm not sure what exception gets thrown here (and if it has an `errors` attribute, but we're already logging the full exception.

File: src/prefect/tasks/gcp/bigquery.py [hunk 4]
```diff
@@ -385,10 +385,21 @@ def run(
...
+                for error in exception.errors:
+                    self.logger.error(error)
+
+            if load_job is not None and load_job.errors is not None:
```
Comment (zangell44) [inline src/prefect/tasks/gcp/bigquery.py:399]: for this to work, we'll need to initalize`load_job = None` before calling `load_job = client.load_table_from_url(...)`

there are two possible sources of errors here
1. an error in initializing `load_job = client.load_table_from_url()`
2. an error in running `load_job` (`load_job.result()`)

if (1) occurs, `load_job` will be undefined, and we'll get a `NameError` here

**Review by zangell44 — CHANGES_REQUESTED** [review]
A few minor feedback points, looking good!

**Comment by joelluijmes** [comment]
Forgot about this PR 🙊 but I've removed the unnneeded for loop 👍

**Review by jcrist — APPROVED** [review]

**Pull request merged**